### DISCUSSION TITLE
Stuck on "Collecting Mesh Info"

### DISCUSSION THREAD

**Discussion by dtlinker** [nextfoam/baram#123] [Q&A] [answered]
I am just starting out with BARAM on a Macbook Pro with M1 chip, running MacOs Sonoma 14.4.1. 

I am stuck with BARAMmesh when generating a mesh. When I click generate, I get a dialog that says "Collecting Mesh Info" that never completes. I have left it overnight but it has no progress. I have it tried two different tutorials, and get stuck in the same place on both.

In case it is important, I also had a problem initially the generation of a mesh which failed, and I finally traced it to the program not finding libmpi.40.dylib even though I had installed it with HomeBrew. I solved that by copying libmpi.40.dylib into Frameworks in BARAMmesh.

Any idea why I am stuck on Collecting Mesh Info and what to do about it?

Thanks

**Comment by jiban** [comment]
Regarding "Collecting Mesh Info" issue, you may have an exception log in a file named "log.txt" in the project folder like following.

![image](https://github.com/user-attachments/assets/a584f18d-ab5d-4e99-9b59-5429c50b1c46)

**Please send me the file.**

And for `libmpi.40.dylib`, BARAM assumes that *Homebrew* folder is `/opt/homebrew/lib` and  `/opt/homebrew/bin`.
Do you have brew files there?

**Reply by dtlinker** [reply]
[log.txt](https://github.com/user-attachments/files/17219955/log.txt)
 Thank you for your response.

I do not have a /opt/homebrew directory.

Here is my brew config:
HOMEBREW_VERSION: 4.3.24
ORIGIN: https://github.com/Homebrew/brew
HEAD: 916044581862c32fc2365e8e9ff0b1507a98925e
Last commit: 10 days ago
Core tap JSON: 25 Sep 07:00 UTC
Core cask tap JSON: 25 Sep 04:55 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local/Homebrew
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.sl4ILDPkvn/org.xquartz:0
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.3.4 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.3.4_1/bin/ruby
CPU: octa-core 64-bit westmere
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.4.1-x86_64
CLT: N/A
Xcode: 15.3
Rosetta 2: true

**Comment by jiban** [comment, marked as answer]
Hi @dtlinker,
As you might notice in the log file, `mpirun` command was not found in the PATH.

I'm a bit confused because you've already said that your machine was **Macbook Pro with M1 chip**.
Here **M1** means [ARM base Apple Silicon](https://en.wikipedia.org/wiki/Apple_M1).
However, your brew setting shows `macOS: 14.4.1-x86_64`. Here **x86_64** means Intel Silicon.

Please refer to [Homebrew Installation](https://docs.brew.sh/Installation) page.
>The script installs Homebrew to its default, supported, best prefix (/opt/homebrew for Apple Silicon, /usr/local for macOS Intel and /home/linuxbrew/.linuxbrew for Linux)

BARAM for macOS supports Apple silicon only for now.
Please check your hardware and brew again.
Thank you.

**Reply by dtlinker** [reply]
Thank you! Very insightful observation. I do have apple silicon (M1) but my system and brew installation was migrated from my old laptop. I re-installed hombrew, and that completely resolved the "Stuck on Collecting Mesh Info" problem. I continued on the tutorial, and the next step "Castellation" now fails.  Please see the console output below.

Thoughts for next steps?

Thanks

/*---------------------------------------------------------------------------*\
| =========                 |                                                 |
| \\      /  F ield         | OpenFOAM: The Open Source CFD Toolbox           |
|  \\    /   O peration     | Version:  2312                                  |
|   \\  /    A nd           | Website:  www.openfoam.com                      |
|    \\/     M anipulation  |                                                 |
\*---------------------------------------------------------------------------*/
Build  : 1716428b-20240530 OPENFOAM=2312 version=v2312
Arch   : "LSB;label=32;scalar=64"
Exec   : /Applications/BaramMesh.app/Contents/Frameworks/solvers/openfoam/bin/snappyHexMesh
Date   : Oct 02 2024
Time   : 14:16:38
Host   : Davids-MacBook-Pro-44.local
PID    : 59512
I/O    : uncollated
Case   : /Users/dtlinker/FDA3/case
nProcs : 1
trapFpe: Floating point exception trapping enabled (FOAM_SIGFPE).
fileModificationChecking : Monitoring run-time modified files using timeStampMaster (fileModificationSkew 0, maxFileModificationPolls 20)
allowSystemOperations : Allowing user-supplied system call operations
// * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * //
Create time
Create mesh for time = 0
Read mesh in = 0.19 s
--> FOAM Warning :
    Reading "system/snappyHexMeshDict" at line 55
    Too many closing '}' ... was a ';' forgotten?
--> FOAM Warning :
    Reading "system/snappyHexMeshDict" at line 268
    Too many closing '}' ... was a ';' forgotten?
--> FOAM Warning :
    Reading "system/snappyHexMeshDict" at line 272
    Too many closing '}' ... was a ';' forgotten?
--> FOAM Warning :
    Reading "system/snappyHexMeshDict" at line 275
    Too many closing ')' ... was a ';' forgotten?
--> FOAM Warning :
    Reading "system/snappyHexMeshDict" at line 276
    Too many closing '}' ... was a ';' forgotten?
--> FOAM Warning :
    Reading "system/snappyHexMeshDict" at line 290
    Too many closing '}' ... was a ';' forgotten?
--> FOAM Warning :
    Reading "system/snappyHexMeshDict" at line 295
    Too many closing '}' ... was a ';' forgotten?
--> FOAM Warning :
    Reading "system/snappyHexMeshDict" at line 313
    Too many closing '}' ... was a ';' forgotten?


--> FOAM FATAL IO ERROR: (openfoam-2312)
Entry 'castellatedMeshControls' not found in dictionary "system/snappyHexMeshDict"


file: system/snappyHexMeshDict at line 10 to 337.

    From static const Foam::dictionary &Foam::meshRefinement::subDict(const dictionary &, const word &, const bool, enum keyType::option)
    in file meshRefinement/meshRefinement.C at line 3812.

FOAM exiting

**Reply by jiban** [reply]
I'm suspecting that your geometry might have non English names.
Could you share your project for me to look into it?

**Reply by dtlinker** [reply]
[FDA3.zip](https://github.com/user-attachments/files/17246696/FDA3.zip)
Of course! I am just trying to work through the tutorial "Axi-symmetric Nozzle" for BARAMMesh.

**Reply by kdg5424** [reply]
Hello @dtlinker. This is Donggyu from NEXTFOAM and a colleague with @jiban.
I found the blank in your boundary name. And the reason why the problem happend was that (The blank in the boundary name).
I deleted blank in boundary name and check that baramMesh operated very well.
So could you try without blank?

![image](https://github.com/user-attachments/assets/a1a3cf95-0a2a-4faf-8996-f6d4156afb2d)

Thank you very much
Best regards

Donggyu Kim

**Reply by dtlinker** [reply]
Thank you so much! That worked perfectly! Now I just need to work on the BARAMFlow tutorial. Thank you again for your very helpful and timely responses.

**Reply by kdg5424** [reply]
You're welcome.

And could you let me know your application?
Maybe we can support or help you for the various ways.
Or you can help and support us haha.
